docs/api/java-api-quickstart.md

```diff
@@ -66,6 +66,17 @@ Table table = catalog.createTable(name, schema, spec);
 
 The logs [schema](#create-a-schema) and [partition spec](#create-a-partition-spec) are created below.
 
+HiveCatalog supports registering the table using an existing metadata JSON file as below:
```

Comment:
Agreed I don't think this should go in the quickstart. Mainly because someone new to Iceberg landing here to try the Java API for the first time won't have an existing metadata file and a configured `HiveCatalog` to register it as a table. I'd say a better place is [java-api.md](https://github.com/apache/iceberg/blob/master/docs/api/java-api.md) which will be with the versioned docs. (Quickstarts will eventually be moved to the common docs)

@ajantha-bhat what do you think?

Reply:
@samredai: I initially thought of keeping in [java-api.md](https://github.com/apache/iceberg/blob/master/docs/api/java-api.md) itself. But there is no catalog related section there. Just adding about `registerCatalog` without background on catalog looked odd to me.

So, I have added here where it has catalog related topics. 

So, I should add some info about catalogs in `java-api.md` and then add this?

Reply:
That's a good point. in my opinion the docs in general are due for some sections that purely focus on catalogs (see figure 2 in this docs site [proposal](https://docs.google.com/document/d/1Y_PRv6p5oJaxg_68AUia_JHw8P4-AZIu3hP5IH2Cpsw/edit) presented at the sync this morning). There's quite a bit there which is why a quick-start probably won't provide enough open space to cover any catalog implementation in detail. At the most, I'd say a quick-start should load an InMemory catalog (once that implementation is ready) and link off to comprehensive catalog docs that cover production grade catalogs.

What do you think of shifting this to a new `hive_catalog.md` file that we can add into the new catalog section when it's ready? I can help structure that page and fill it with other content that I think is missing right now.

Reply:
As you mentioned in the proposal, definitely having one doc per each catalog makes sense.

Again, now keeping just this PR change in a new file `hive_catalog.md` looks odd to me.

a) I would rather have this merged in quick-start now and refactor later when separate catalog page is ready.
b) Or keep this PR open till separate catalog page is ready and move this change on top of it.